File: Leetcode 0187 Repeated DNA Sequences.go
// 2026/08/17
func findRepeatedDnaSequences(s string) []string {
    seen := make(map[string]bool)
    repeated := make(map[string]bool)
    var ans []string
    if len(s) < 11{
        return ans
    }

    for i:=0; i+10<=len(s); i++{
        sub := s[i:i+10]
        if seen[sub]{
            repeated[sub] = true
        }
        seen[sub] = true
    }
    
    for i := range repeated{
        ans = append(ans, i)
    }

    return ans

}

File: Leetcode 0187 Repeated DNA Sequences.py
# 2025/08/17
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        seen = set()
        ans = set()
        if len(s) < 11:
            return []
        for i in range(0, len(s)-9):
            num = s[i:i+10]
            if num in seen:
                ans.add(num)
            seen.add(num)
            
        return list(ans)
